provider/provider.go

```diff
@@ -125,7 +125,10 @@ func (p *dockerNativeProvider) Check(ctx context.Context, req *rpc.CheckRequest)
 	}
 
 	// add implicit resource to provider
-	inputs["build"].ObjectValue()["contextDigest"] = resource.NewStringProperty(contextDigest)
+	inputs["build"] = resource.NewObjectProperty(resource.PropertyMap{
```

Comment:
This would replace `inputs["build"]` if the user set it - are we overriding anything the user would input?

Reply:
d'oh!

Added a null check condition.